src/main/java/org/opentripplanner/transit/model/network/TripPattern.java

```diff
@@ -402,7 +402,12 @@ public TripPattern getOriginalTripPattern() {
   }
 
   public I18NString getTripHeadsign() {
-    var tripTimes = scheduledTimetable.getRepresentativeTripTimes();
+    var representativeTripTimes = scheduledTimetable.getRepresentativeTripTimes();
+    var tripTimes = representativeTripTimes == null &&
+      originalTripPattern != null &&
```

Comment:
It's a bit hard to read these three boolean expression. can you please extract them in a method and give it a good name?

Other wise this looks solid.

Reply:
Added a couple of commits which splits this into methods